### DISCUSSION TITLE
prevent grapesjs from adding additional styling to imported content

### DISCUSSION THREAD

**Discussion by KaoDeo** [GrapesJS/grapesjs#5252] [Q&A] [answered]
Hi guys, 

I have a user case to share. Some of our users are currently importing content that was created in another email editor provider since the GrapesJS feature in our app is still new, and they haven't fully migrated to it yet. Turned out that there is a difference between the content user sees in the editor and the  appearance of the emails in the actual environment after they are sent. We have tested it and seems that GrapesJS adds some additional styling to the imported content, resulting in this difference. I might be wrong about how GrapesJS handles imported content, however I am sure that the problem resides in the editor (my wrong code or potential bug in the editor itself) as we have sent the content from the email provider directly and there was no disparity.

I thought a potential solution to this issue would be to find a way for GrapesJS to prevent adding any styling to the imported content. Is there a way to achieve this? Or do you know more about internal workings of the handling the content in general which might help me solve this issue?

Thanks!

**Comment by artf** [comment]
Hi @KaoDeo the editor should import what is asked to import so it's not clear what are the additional styling you're referring to. It's also hard to have any idea of what is happening without a reproducible demo.

**Reply by KaoDeo** [reply]
okey, here is the demo newsletter which was different in editor and real environment. Below you can see the input code which I imported to grapesjs(the code was generated by another email editor), the logic which combined the content and the final output which was sent to the server. 

To make sure that the problem was with grapesjs, I sent the the same content directly from Sendgrid email editor and it was delivered correctly. As users were really concerned, I imported Monaco editor as a quick solution, where they pasted their html, viewed the preview and sent it aand that was also delivered correctly. However, I don't really like that solution and would prefer to solve this issue.

**input code**: https://codepen.io/kaodeo/pen/rNQqMJq

**logic:**

```
    compileHtml(html: string, styles: string, script: string) {
    const regexForIndex = /(<\/body>)/g;
    const everyLastBodyTagIndex = html.search(regexForIndex);
    const result =
      `<head><style>${styles}</style></head>` +
      html.slice(0, everyLastBodyTagIndex) +
      `<script>${script}</script>` +
      html.slice(everyLastBodyTagIndex);

    return result;
  }

  private buildContentModel(): HtmlTemplateEditorContentModel {
    const html = this.editor.getHtml();
    const styles = this.editor.getCss() as string;
    const script = this.editor.getJs();

    return {
      content: this.compileHtml(html, styles, script), 
    };
  }

```  

**output code**: https://codepen.io/kaodeo/pen/eYQPdjP

**Content in the editor vs content in web Gmail:**
<img width="794" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/91371368/2f1acb50-e878-4272-a3ce-8ccd729a8876">
<img width="786" alt="image" src="https://github.com/GrapesJS/grapesjs/assets/91371368/499e94a8-25d1-419a-815a-a22b80073967">

**Reply by artf** [reply, marked as answer]
@KaoDeo unfortunately importing the entire HTML document doesn't produce the same output (you can see how some of the head elements are inserted in the body) as the input HTML string and that's definitely might be a problem for email clients.

You can probably update your export method to handle those cases but, at the moment, I don't feel the urgency to focus on it right now.
